esp/esdllib/esdl_script.cpp

```diff
@@ -0,0 +1,914 @@
+{
+    LogLevel level = LogMin;
+    if (!transforms.size())
+        return;
+    if (tgtcfg)
+        level = (unsigned) tgtcfg->getPropInt("@traceLevel", level);
+
+    if (request.length()!=0)
+    {
+        if (level >= LogMax)
+        {
+            DBGLOG("ORIGINAL REQUEST: %s", request.str());
+            StringBuffer marshalled;
+            if (bindingCfg)
+                toXML(bindingCfg, marshalled.clear());
+            DBGLOG("BINDING CONFIG: %s", marshalled.str());
+            if (tgtcfg)
+                toXML(tgtcfg, marshalled.clear());
+            DBGLOG("TARGET CONFIG: %s", marshalled.str());
+        }
+
+        bool strictParams = bindingCfg ? bindingCfg->getPropBool("@strictParams", false) : false;
+        Owned<IXpathContext> xpathContext = getXpathContext(request.str(), strictParams);
+
+        StringArray prefixes;
+        for ( IEsdlCustomTransform * const & item : transforms)
+        {
+            if (item)
+                item->appendEsdlURIPrefixes(prefixes);
+        }
+
+        registerEsdlXPathExtensions(xpathContext, context, prefixes);
+
+        VStringBuffer ver("%g", context->getClientVersion());
+        if(!xpathContext->addInputValue("clientversion", ver.str()))
+            OERRLOG("Could not set custom transform variable: clientversion:'%s'", ver.str());
+
+        //in case transform wants to make use of these values:
+        //make them few well known values variables rather than inputs so they are automatically available
+        xpathContext->addVariable("query", tgtcfg->queryProp("@queryname"));
+        xpathContext->addVariable("method", method);
+        xpathContext->addVariable("service", service);
+        xpathContext->addVariable("request", reqtype);
+
+        ISecUser *user = context->queryUser();
+        if (user)
+        {
+            static const std::map<SecUserStatus, const char*> statusLabels =
+            {
+#define STATUS_LABEL_NODE(s) { s, #s }
+                STATUS_LABEL_NODE(SecUserStatus_Inhouse),
+                STATUS_LABEL_NODE(SecUserStatus_Active),
+                STATUS_LABEL_NODE(SecUserStatus_Exempt),
+                STATUS_LABEL_NODE(SecUserStatus_FreeTrial),
+                STATUS_LABEL_NODE(SecUserStatus_csdemo),
+                STATUS_LABEL_NODE(SecUserStatus_Rollover),
+                STATUS_LABEL_NODE(SecUserStatus_Suspended),
+                STATUS_LABEL_NODE(SecUserStatus_Terminated),
+                STATUS_LABEL_NODE(SecUserStatus_TrialExpired),
+                STATUS_LABEL_NODE(SecUserStatus_Status_Hold),
+                STATUS_LABEL_NODE(SecUserStatus_Unknown),
+#undef STATUS_LABEL_NODE
+            };
+
+            Owned<IPropertyIterator> userPropIt = user->getPropertyIterator();
+            ForEach(*userPropIt)
+            {
+                const char *name = userPropIt->getPropKey();
+                if (name && *name)
+                    xpathContext->addVariable(name, user->getProperty(name));
```

Comment:
I believe this is intended to be a call to addInputValue so that an explicit param declaration is required when strictParams is enabled.

Reply:
Good catch.  Missed this when I changed the others.